discovery/gossiper.go

```diff
@@ -2029,22 +2029,37 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []n
 				remotePub, _ = chanInfo.NodeKey1()
 			}
 
-			sPeer, err := d.cfg.FindPeer(remotePub)
-			if err != nil {
-				log.Errorf("unable to send channel update -- "+
-					"could not find peer %x: %v",
-					remotePub, err)
-			} else {
-				// Send ChannelUpdate directly to remotePeer.
-				// TODO(halseth): make reliable send?
-				err = sPeer.SendMessage(false, msg)
-				if err != nil {
-					log.Errorf("unable to send channel "+
+			// Now, we'll attempt to send the channel update message
+			// reliably to the remote peer in the background, so
+			// that we don't block if the peer happens to be offline
+			// at the moment.
+			d.wg.Add(1)
```

Comment:
We also need to trigger something like this on start up. Consider a case where a series of restarts happen (on either side) before the channel update is ultimately sent. We already so something similar for sending the `AnnouncementSig` message, so perhaps we can extend that to also handle this case.